### PR TITLE
Kops spec

### PR DESCRIPTION
**Pull request by sh31**
__kops spec__
* Create Terraform v0.12 kops spec module
* Get global remote state
* Create Kops Spec
* Change t2 machines to t3 & r4 machines to r5
* Create script to create/update Kops cluster

### CODE REVIEW

**Comment by sh31** [comment]
Merge module first. Then rebase this branch.

File: modules/kops_spec/main.tf [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    taints                    = "[]"
+    node_labels               = "{}"
+  }
+}
+
+data "template_file" "kops" {
+  template = "${file("${path.module}/templates/kops.yaml.tmpl")}"
+
+  vars = {
+    cluster_dns_name = "${var.cluster_dns_name}"
```
Comment (djsd123) [inline modules/kops_spec/main.tf]: Any particular reason why the variable calls are inconsistent?

File: platform-base/kops.sh [hunk 2]
```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+temp_role=$(aws sts assume-role \
+                    --role-arn $ROLE_ARN \
+                    --role-session-name "codebuild-assume-kops-script")
+
+export AWS_ACCESS_KEY_ID=$(echo $temp_role | jq .Credentials.AccessKeyId | xargs)
```
Comment (djsd123) [inline platform-base/kops.sh]: If these variables do not get set, I assume the script fails?

If so, then you may want to add some bash options.  Namely `set -u` and maybe protect against pipe fails. `set -o pipefail`

File: platform-base/variables.tf [hunk 3]
```diff
@@ -90,15 +90,15 @@ variable "k8s_availability_zones" {
 }
 
 variable "k8s_masters_machine_type" {
-  default = "t2.medium"
+  default = "t3.medium"
```
Comment (djsd123) [inline platform-base/variables.tf]: Nice work 👍

File: modules/kops_spec/variables.tf [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+variable "highmem_nodes_root_volume_size" {}
+
+variable "bastions_extra_sg_id" {}
+variable "bastions_machine_type" {}
+variable "bastions_instancegroup_max_size" {}
+variable "bastions_instancegroup_min_size" {}
+variable "bastions_root_volume_size" {}
+
+variable "kops_state_bucket" {}
+variable "oidc_client_id" {}
```
Comment (djsd123) [inline modules/kops_spec/variables.tf]: On a separate note.  You'll need to create an application

**Comment by djsd123** [comment]
The bastion appears to be on the wrong network

File: platform-base/data.tf [hunk 5]
```diff
@@ -1 +1,11 @@
 data "aws_caller_identity" "current" {}
+
+data "terraform_remote_state" "global" {
+  backend = "s3"
+
+  config = {
+    bucket = "${var.terraform_bucket_name}"
```
Comment (djsd123) [inline platform-base/data.tf]: Here too mate.  You need a decent terraform editor 😉

**Comment by djsd123** [comment]
Can you update the `oidc_client_id` for prod & dev?  Kubernetes-API is the client you want.

**Comment by djsd123** [comment]
Can you rebase, please?

**Pull request merged**